### PR TITLE
Support replacing external scripts with local copies from disk

### LINKED ISSUE #26456: Support loading unminified JS files from disk instead of network
Running with `--unminify-js` gives us a directory tree on disk that effectively mirrors the network, but with more readable source files. If we stored the HTTP response headers alongside the actual unminified files, we could theoretically avoid the network and read directly from disk when the files exist. This would then allow us to add debugging code to the scripts to help track down errors when we don't have access to the original source.

To accomplish this, we will need to:
* add a new command line argument that enables checking a given directory before going to the network
* store HTTP response headers as part of --unminify-js
* make the --unminify-js mode create directory trees that mirror the entire path of the URL
* only use the disk mode for GET requests
* rewrite any Content-Length and Content-Encoding headers for disk-based responses appropriately

### PR DESCRIPTION
**Pull request by skrzyp1**
---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #26456(GitHub issue number if applicable)

- [ ] There are tests for these changes OR
- [x] These changes do not require tests because ___

### CODE REVIEW

File: components/script/dom/window.rs [hunk 1]
```diff
@@ -1981,8 +1984,8 @@ impl Window {
         if let Some(&Host::Domain(ref host)) = document.url().origin().host() {
             let mut path = env::current_dir().unwrap();
             path.push("unminified-js");
-            path.push(host);
-            let _ = fs::remove_dir_all(&path);
+            //path.push(host);
+            //let _ = fs::remove_dir_all(&path);
             match fs::create_dir_all(&path) {
                 Ok(_) => {
                     *self.unminified_js_dir.borrow_mut() =
```
[the remark this answers is not in the document]
Comment (jdm) [reply components/script/dom/window.rs:1991]: I agree. Let's remove this code.

**Review by jdm — CHANGES_REQUESTED** [review]
This looks really good so far! Please squash all of the commits into one, then rebase that commit against master to remove any merge conflicts.

File: components/script/dom/window.rs [hunk 2]
```diff
@@ -1981,8 +1984,8 @@ impl Window {
         if let Some(&Host::Domain(ref host)) = document.url().origin().host() {
             let mut path = env::current_dir().unwrap();
             path.push("unminified-js");
-            path.push(host);
-            let _ = fs::remove_dir_all(&path);
+            //path.push(host);
+            //let _ = fs::remove_dir_all(&path);
```
Comment (CYBAI) [inline components/script/dom/window.rs:1988]: maybe should be removed if unused 👀?

File: components/script/dom/window.rs [hunk 3]
```diff
@@ -1993,7 +1996,7 @@ impl Window {
                         self.unminified_js_dir.borrow()
                     );
                 },
-                Err(_) => warn!("Could not create unminified dir for {:?}", host),
+                Err(_) => warn!("Could not create unminified dir"), // for {:?}", host),
```
Comment (CYBAI) [inline components/script/dom/window.rs:1999]: ditto.

```suggestion
                Err(_) => warn!("Could not create unminified dir"),
```

**Pull request merged**